From 87b9c88a74eb2247c9b87ecfcf036be6fd2925d2 Mon Sep 17 00:00:00 2001
From: Fancy Fang <chen.fang@freescale.com>
Date: Fri, 29 Aug 2014 15:21:54 +0800
Subject: [PATCH] ENGR00329330 ARM: IMX6SX: PXP: thread wait queue should be
 init before thread creation

Since the thread function will use the thread wait queue
as soon as the thread is created.

Signed-off-by: Fancy Fang <chen.fang@freescale.com>
(cherry picked from commit 325544504ea2bd73749ef65064be0b4ab885b987)
---
 drivers/dma/pxp/pxp_dma_v2.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/dma/pxp/pxp_dma_v2.c b/drivers/dma/pxp/pxp_dma_v2.c
index d5b2d8f..72badd1 100644
--- a/drivers/dma/pxp/pxp_dma_v2.c
+++ b/drivers/dma/pxp/pxp_dma_v2.c
@@ -1722,13 +1722,13 @@ static int pxp_probe(struct platform_device *pdev)
 	pxp->clk_timer.function = pxp_clkoff_timer;
 	pxp->clk_timer.data = (unsigned long)pxp;
 
+	init_waitqueue_head(&pxp->thread_waitq);
 	/* allocate a kernel thread to dispatch pxp conf */
 	pxp->dispatch = kthread_run(pxp_dispatch_thread, pxp, "pxp_dispatch");
 	if (IS_ERR(pxp->dispatch)) {
 		err = PTR_ERR(pxp->dispatch);
 		goto exit;
 	}
-	init_waitqueue_head(&pxp->thread_waitq);
 	tx_desc_cache = kmem_cache_create("tx_desc", sizeof(struct pxp_tx_desc),
 					  0, SLAB_HWCACHE_ALIGN, NULL);
 	if (!tx_desc_cache) {
-- 
1.8.0

